package.json

```diff
@@ -106,5 +122,11 @@
       "@semantic-release/npm",
       "@semantic-release/git"
     ]
+  },
+  "lint-staged": {
+    "src/**/*.{ts,tsx}": [
+      "yarn lint:fix",
```

Comment:
You normally don't want to use your `lint` scripts as those can lint the whole project with a glob pattern. This could nullify the effect of `lint-staged`. You just have to add the scripts you want to run and `lint-staged` determines which files it will be on. This line would change to

```
"tsc --noEmit && tslint --fix",
```

Reply:
see above 👆 :)